main.go

```diff
@@ -1190,6 +1187,13 @@ func startDRL() {
 	startRateLimitNotifications()
 }
 
+// The only purpose is to allow mainRouter to be dynamically replaced
+type appHandler struct{}
```

Comment:
I'd call this `mainHandler` to clarify that it's always linked to `mainRouter`.

Also, please use the godoc format as in `mainHandler's only purpose is...`